url5583f3dbb2_XOR19_XScript_tgz-pJ8-xscript_interactive_MainJ8/src/xscript/XExec.java

```diff
@@ -0,0 +1,1927 @@
+			} else {
+				callBiOperator(">=", "__greaterEqual__", "__greaterEqual2__", v2, v1);
+			}
+			break;
+		case IDIV:
+			v1 = pop();
+			v2 = pop();
+			if (v1.isNumber() && v2.isNumber()) {
+				if (v1.isFloat() || v2.isFloat()) {
+					v3 = XValueInt.valueOf((long) (v2.getFloat() / v1.getFloat()));
+				} else {
+					v3 = XValueInt.valueOf(v2.getInt() / v1.getInt());
+				}
+				push(v3);
+			} else {
+				callBiOperator("\\", "__idiv__", "__idiv2__", v2, v1);
+			}
+			break;
+		case IMPORT:
+			s = callFrame.readStringP();
+			v1 = rt.alloc(s);
+			callHelper("__import", v1);
+			break;
+		case IMPORT_SAVE:
+			s = callFrame.readStringP();
+			v1 = pop();
+			v2 = pop();
+			XUtils.putAtMapItem(rt, v1, s, v2);
+			break;
+		case INC:
+			v1 = pop();
+			if (v1.isNumber()) {
+				if (v1.isFloat()) {
+					v2 = new XValueFloat(v1.getFloat() + 1);
+				} else {
+					v2 = XValueInt.valueOf(v1.getInt() + 1);
+				}
+				push(v2);
+			} else {
+				callSingleOperator("++", "__inc__", v1);
+			}
+			break;
+		case INSTANCEOF:
+			v1 = pop();
+			v2 = pop();
+			v3 = XValueBool.valueOf(XUtils.isInstanceOf(rt, v2, v1));
+			push(v3);
+			break;
+		case INVERT:
+			v1 = pop();
+			if (v1.isInt()) {
+				v2 = XValueInt.valueOf(~v1.getInt());
+				push(v2);
+			} else if (v1.isBool()) {
+				v2 = XValueBool.valueOf(!v1.getBool());
+				push(v2);
+			} else {
+				callSingleOperator("~", "__invert__", v1);
+			}
+			break;
+		case ISHR:
+			v1 = pop();
+			v2 = pop();
+			if (v1.isInt() && v2.isInt()) {
+				v3 = XValueInt.valueOf(v2.getInt() >>> v1.getInt());
+				push(v3);
+			} else {
+				callBiOperator(">>>", "__ishr__", "__ishr2__", v2, v1);
+			}
+			break;
+		case ISDERIVEDOF:
+			v1 = pop();
+			v2 = pop();
+			v3 = XValueBool.valueOf(XUtils.isDerivedOf(rt, v2, v1));
+			push(v3);
+			break;
+		case JUMP:
+			callFrame.jumpTo(callFrame.readUShort());
+			break;
+		case JUMP_IF_NON_ZERO:
+			i1 = callFrame.readUShort();
+			v1 = pop();
+			if (v1.noneZero()) {
+				callFrame.jumpTo(i1);
+			}
+			break;
+		case JUMP_IF_ZERO:
+			i1 = callFrame.readUShort();
+			v1 = pop();
+			if (!v1.noneZero()) {
+				callFrame.jumpTo(i1);
+			}
+			break;
+		case LADD:
+			v1 = pop();
+			v2 = pop();
+			if (v1.isNumber() && v2.isNumber()) {
+				if (v1.isFloat() || v2.isFloat()) {
+					v3 = new XValueFloat(v2.getFloat() + v1.getFloat());
+				} else {
+					v3 = XValueInt.valueOf(v2.getInt() + v1.getInt());
+				}
+				push(v3);
+			} else {
+				callBiOperator2("+", "__ladd__", "__ladd2__", "__add__", "__add2__", v2, v1);
+			}
+			break;
+		case LAND:
+			v1 = pop();
+			v2 = pop();
+			if ((v1.isInt() || v1.isBool()) && (v2.isInt() || v2.isBool())) {
+				if (v1.isBool() && v2.isBool()) {
+					v3 = XValueBool.valueOf(v2.getBool() && v1.getBool());
+				} else {
+					v3 = XValueInt.valueOf(v2.getInt() & v1.getInt());
+				}
+				push(v3);
+			} else {
+				callBiOperator2("&", "__land__", "__land2__", "__and__", "__and2__", v2, v1);
+			}
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Similar code fragments were detected in the same file at the following lines: 1041:1050, 1195:1204, 1257:1266.  
Refactoring can help improve code maintainability. Consider reducing duplicate code by extracting it into a separate method. You can then replace duplicated code with calls to this new method.